medical-system/src/pages/PatientsPage.tsx

```diff
@@ -0,0 +1,51 @@
+import { useState } from "react";
+import { Modal } from "../components/shared/Modal";
+import { Table, User } from "../components/shared/Table";
+import useSWR from "swr";
+import { addDoc, collection } from "firebase/firestore";
+import { db } from "../auth/firebase-config";
+
+export function PatientsPage() {
+  const [modalOpen, setModalOpen] = useState(false);
+  const [newPatient, setNewPatient] = useState<User | null>(null);
+
+  const postPatients = async () => {
+    try {
+      const patientsCollectionRef = collection(db, "patients");
+      const docRef = await addDoc(patientsCollectionRef, newPatient);
+
+      console.log("Patient added successfully");
+    } catch (error) {
+      console.log("Error adding patient:", error);
+    }
+  };
+
+  const { mutate } = useSWR("patients", postPatients);
+
+  const handleAddPatient = (patient: User) => {
+    setNewPatient(patient);
+    postPatients();
+    mutate();
```

Comment:
https://medium.com/@jayashakthiperera/power-of-swr-with-react-useswrmutation-61a0aa82d958